### PR TITLE
Integrate date time parser into assign command

### PR DESCRIPTION
**Pull request by JinHao-L**
Added parser for date time to recognise natural datetime language and formatted datetime
Integrated date time parser into Assign related command and tests

#78

### CODE REVIEW

File: src/main/java/seedu/address/model/appointment/Time.java [hunk 1]
```diff
@@ -20,8 +21,10 @@
     // Closes at 10pm
     public static final LocalTime CLOSING_TIME = LocalTime.of(22, 0);
 
-    public static final String MESSAGE_CONSTRAINTS = "The time should fall within the opening hours: "
-            + "from " + OPENING_TIME.format(TIME_FORMAT) + " to " + CLOSING_TIME.format(TIME_FORMAT);
+    public static final String MESSAGE_CONSTRAINTS = "The appointment time period should fall within "
+            + "the opening hours: from " + OPENING_TIME.format(TIME_FORMAT)
+            + " to " + CLOSING_TIME.format(TIME_FORMAT)
+            + "\nThe appointment duration of " + DEFAULT_DURATION.toHours() + "hours should also be considered.";
```
Comment (xz0127) [inline src/main/java/seedu/address/model/appointment/Time.java]: If the default duration is 1 hour, should "hour" instead of "hours" be used?

**Review by xz0127 — COMMENTED** [review]
Other than the minor comment, the rest LGTM! 👍

File: src/main/java/seedu/address/model/appointment/Time.java (hunk 1 above)
Comment (JinHao-L) [reply src/main/java/seedu/address/model/appointment/Time.java]: Oh I missed that out! I'll update it to "hour". Thanks 👍

**Review by xz0127 — APPROVED** [review]
LGTM! 👍

**Pull request merged**